src/solvers/flattening/arrays.cpp

```diff
@@ -185,9 +189,10 @@ void arrayst::collect_arrays(const exprt &a)
   }
   else if(a.id()==ID_member)
   {
-    if(to_member_expr(a).struct_op().id()!=ID_symbol)
-      throw
-        "unexpected array expression: member with `"+a.op0().id_string()+"'";
+    DATA_INVARIANT(
+      to_member_expr(a).struct_op().id()==ID_symbol,
+      ("unexpected array expression: member with `"+
+       a.op0().id_string()+"'").c_str());
```

Comment:
[Not an issue for this PR, my problem not yours] This looks a little clunky.  Do you think we could improve invariant.h to de-clunky this?